### PR TITLE
Xcode9 support

### PR DESCRIPTION
**Pull request by vytis**
# Changes

### Implementation
* Warning about `Void` in closures
* `PowerProvider` had method with old-style method name. Xcode9 added warning for protocol methods that are not implemented 
* Ambigous selector warning for location message cell actions
* Redundant conformances to `Equatable`

### Tests
* Fix failures because we were treating phone width `NSNumber` as `NSValue`. This seemed to be working earlier, but now returns 0 as width and all snapshots fail.

### Dependencies
Dependency | Branch | PR      | CI Status
------------- | ------ | ------------- | --------
[wire-ios-share-engine](https://github.com/wireapp/wire-ios-share-engine) | [feature/xcode9](https://github.com/wireapp/wire-ios-share-engine/compare/feature/xcode9) | No changes | [CircleCI](https://circleci.com/gh/wireapp/wire-ios-share-engine/tree/feature%2Fxcode9)
[wire-ios-sync-engine](https://github.com/wireapp/wire-ios-sync-engine) | [feature/xcode9](https://github.com/wireapp/wire-ios-sync-engine/compare/feature/xcode9) |  https://github.com/wireapp/wire-ios-sync-engine/pull/583 | [CircleCI](https://circleci.com/gh/wireapp/wire-ios-sync-engine/tree/feature%2Fxcode9)
[wire-ios-canvas](https://github.com/wireapp/wire-ios-canvas) | [feature/xcode9](https://github.com/wireapp/wire-ios-canvas/compare/feature/xcode9) | No changes | [CircleCI](https://circleci.com/gh/wireapp/wire-ios-canvas/tree/feature%2Fxcode9)
[wire-ios-ziphy](https://github.com/wireapp/wire-ios-ziphy) | [feature/xcode9](https://github.com/wireapp/wire-ios-ziphy/compare/feature/xcode9) | No changes | [CircleCI](https://circleci.com/gh/wireapp/wire-ios-ziphy/tree/feature%2Fxcode9)

### CODE REVIEW

File: Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme [hunk 1]
```diff
@@ -26,6 +26,7 @@
       buildConfiguration = "Debug"
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
+      language = ""
```
Comment (typfel) [inline Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme]: Intentional?

Comment (vytis) [reply Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme]: It seems Xcode sets this thing after opening the project automatically...

Comment (typfel) [reply Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme]: Ok, I guess it's fine then :)

File: Wire-iOS/Sources/UserInterface/Location/NSUserDefaults+Location.swift [hunk 2]
```diff
@@ -23,10 +23,6 @@ private let latitudeKey = "LastLocationLatitudeKey"
 private let longitudeKey = "LastLocationLongitudeKey"
 private let zoomLevelKey = "LastLocationZoomLevelKey"
 
-extension Dictionary where Key: Hashable, Value: Any {
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Location/NSUserDefaults+Location.swift]: This used to declare that the Dictionary where Key: Hashable, Value: Any is already perfect, it has nothing to add.

Comment (vytis) [reply Wire-iOS/Sources/UserInterface/Location/NSUserDefaults+Location.swift]: It didn't really add any methods, so not completely sure what was the reason behind it

**Pull request merged**